patroni/validator.py

```diff
@@ -38,8 +38,14 @@ def validate_host_port(host_port, listen=False, multiple_hosts=False):
         else:
             hosts = [hosts]
         for host in hosts:
-            proto = socket.getaddrinfo(host, "", 0, socket.SOCK_STREAM, 0, socket.AI_PASSIVE)
-            s = socket.socket(proto[0][0], socket.SOCK_STREAM)
+            if host == "*":
+                if len(hosts) != 1:
+                    raise ConfigParseError("expecting '*' alone")
+                host = ""
+                s = socket.socket(type=socket.SOCK_STREAM)
```

Comment:
In case if hosts is set to `['*']` we need to find all possible IP's to listen on using:
```python
socket.getaddrinfo(None, port, 0, socket.SOCK_STREAM, 0, socket.AI_PASSIVE)
```

It returns the list of  `(family, socktype, proto, canonname, sockaddr)`:
```
[
    (2, 1, 6, '', ('0.0.0.0', 8080)),
    (10, 1, 6, '', ('::', 8080, 0, 0))
]
```

Where `'0.0.0.0'` and `'::'` are IPs.
These detected IPs should be put into the `hosts` list. It all should happen outside of `for host in hosts` loop.

Reply:
Changed accordingly; thanks.